Public/Src/FrontEnd/Nuget/NugetResolver.cs

```diff
@@ -62,12 +69,87 @@ public override async Task<bool> InitResolverAsync(IResolverSettings resolverSet
 
             m_owningModules = new Dictionary<ModuleId, Package>();
 
-            foreach (var package in maybePackages.Result)
+            foreach (var package in maybePackages.Result.Values.SelectMany(v => v))
             {
                 m_packages[package.Id] = package;
                 m_owningModules[package.ModuleId] = package;
             }
 
+            if (Configuration.FrontEnd.GenerateCgManifestForNugets.IsValid ||
+                Configuration.FrontEnd.ValidateCgManifestForNugets.IsValid)
+            {
+                var cgManfiestGenerator = new NugetCgManifestGenerator(Context);
+                string generatedCgManifest = cgManfiestGenerator.GenerateCgManifestForPackages(maybePackages.Result);
+                string existingCgManifest = "{}";
+
+                if ( !Configuration.FrontEnd.GenerateCgManifestForNugets.IsValid &&
+                      Configuration.FrontEnd.ValidateCgManifestForNugets.IsValid )
+                {
+                    // Validation of existing cgmainfest.json results in failure due to mismatch. Should fail the build in this case.

```

Comment:
>cgmainfest.json [](start = 46, length = 15)

the filename is user configurable so it might be better to just refer to it as the manifest